### DISCUSSION TITLE
How to catch network errors

### DISCUSSION THREAD

**Discussion by Maxou44** [katspaugh/wavesurfer.js#3229] [Q&A] [answered]
Hello,

This library is really amazing and work without any issues 👌 

I have a little issue on my app, if the user lost Internet connection (or any network issues), the fetch request of Wavesurfer fails and throw an error. I tried to put a try/catch statement around the init but it don't work and continue to throw an error.

How can I detect and intercept network errors so as to manage them correctly in the app?

Thanks for your help,
Maxime

**Comment by katspaugh** [comment]
You should use the load method which returns a promise.

**Comment by Maxou44** [comment]
I tried the following code :

```js
 try {
    waveSurferRef = WaveSurfer.create({
        container: domRef.current,
        waveColor: inactiveColor,
        progressColor: activeColor,
        cursorColor: 'transparent',
        autoplay: false,
        normalize: true,
        hideScrollbar: true,
        dragToSeek: true,
        autoCenter: true,
        barHeight: 0.5,
        height: 56,
    })
    waveSurferRef.on('interaction', () => {
        waveSurferRef.play();
        //setIsPlaying(true);
        //onPlay(id);
    });
    waveSurferRef.on('finish', () => {
        //setIsPlaying(false);
        //waveSurferRef.media.currentTime = 0;
    });
    waveSurferRef.load('[LINK TO THE AUDIO RESOURCE]').catch(console.warning);
  
  } catch (err) {
    console.warn('Waveform error', err);
  }
```

And I still have the following error :

![image](https://github.com/katspaugh/wavesurfer.js/assets/8138585/99d4e85e-6bef-49f4-9bad-11eb6c8faecb)

From this line:

![image](https://github.com/katspaugh/wavesurfer.js/assets/8138585/b03749a7-d8fb-434d-961c-993f949b9401)

Any ideas?

**Reply by katspaugh** [reply, marked as answer]
Like I said, load returns a promise. So you have to catch it using .catch(), or use async-await.
